app/models/manageiq/providers/ansible_runner_workflow.rb

```diff
@@ -40,41 +40,53 @@ def execute
     response = launch_runner
 
     if response.nil?
-      queue_signal(:abort, "Failed to run ansible #{execution_type}", "error")
+      route_signal(:abort, "Failed to run ansible #{execution_type}", "error")
     else
       context[:ansible_runner_response] = response.dump
 
       started_on = Time.now.utc
       update!(:context => context, :started_on => started_on)
       miq_task.update!(:started_on => started_on)
 
-      queue_signal(:poll_runner)
+      route_signal(:poll_runner)
     end
   end
 
   def poll_runner
-    response = Ansible::Runner::ResponseAsync.load(context[:ansible_runner_response])
-    if response.running?
-      if started_on + options[:timeout] < Time.now.utc
-        response.stop
-
-        queue_signal(:abort, "ansible #{execution_type} has been running longer than timeout", "error")
+    loop do
```

Comment:
I feel like this logic has become spaghetti code as a result.  Seems like breaking this method up instead of trying to implement the `loop` construct for both cases (when only one case needs it) would be wiser:

```ruby
def poll_runner
  MiqEnvironment::Command.is_podified? ? wait_for_response : wait_for_response_async
end

private

def wait_for_response
  response = fetch_response

  while response.running?
    sleep options[:poll_interval]
    response = fetch_response
  end
  
  handle_response(response)
end

def wait_for_response_async
  response = fetch_response

  if response.running?
    queue_signal(:poll_runner, :deliver_on => deliver_on)
  else
    handle_response(response)
  end
end

def fetch_response(response)
  Ansible::Runner::ResponseAsync.load(context[:ansible_runner_response])
end

def handle_response(response)
  result = response.response

  context[:ansible_runner_return_code] = result.return_code
  context[:ansible_runner_stdout]      = result.parsed_stdout

  if result.return_code != 0
    set_status("ansible #{execution_type} failed", "error")
    _log.warn("ansible #{execution_type} failed:\n#{result.parsed_stdout.join("\n")}")
  else
    set_status("ansible #{execution_type} completed with no errors", "ok")
  end
  route_signal(:post_execute)
end
```

Reply:
You might be able to DRY this up a bit more than I have done here, but this was the best that I could come up with off the top of my head.

Reply:
Thatks, yeah I'll put something together. That looks way better

Reply:
Also will probably try to clarify the `response` naming here. `result = response.response` has me cringing every time I see it.

Reply:
The update came out really nice...much easier to read.